src/prognostic_equations/edmfx_sgs_flux.jl

```diff
@@ -482,19 +482,42 @@ function edmfx_sgs_diffusive_flux_tendency!(
             top = Operators.SetValue(C3(FT(0))),
             bottom = Operators.SetValue(C3(FT(0))),
         )
-        MatrixFields.unrolled_foreach(cloud_tracers) do (ρχ_name, χ_name)
+        n = n_mass_flux_subdomains(turbconv_model)
+
+        MatrixFields.unrolled_foreach(cloud_tracers) do (ρχ_name, χ_name, χʲ_name)
             MatrixFields.has_field(Y, ρχ_name) || return
-            ᶜρχ = MatrixFields.get_field(Y, ρχ_name)
-            @. ᶜρχₜ_diffusion = ᶜdivᵥ_ρq(-(ᶠρaK_h * ᶠgradᵥ(specific(ᶜρχ, Y.c.ρ))))
+
             ᶜρχₜ = MatrixFields.get_field(Yₜ, ρχ_name)
+            for j in 1:n
```

Comment:
This wouldn't work for multiple updrafts as you have e.g. `@name(c.sgsʲs.:(1).q_liq)` before. Just add `@assert n_prognostic_mass_flux_subdomains(turbconv_model) == 1` and remove the loop?